windows/utils/win_get_file_folder.yml

```diff
@@ -3,29 +3,43 @@
 ---
 # Get specified path from Windows guest OS to local
 # Parameters:
-#   win_get_src_path: the source path
-#   win_get_dst_path: the destination path
+#   win_get_src_path: the source file or folder path in guest OS.
+#   win_get_dst_path: the destination path on local machine.
 #
-- name: "Set the fact of file path"
+- name: "Set facts of the source path and destination path"
   ansible.builtin.set_fact:
     win_src_path: "{{ win_get_src_path }}"
     win_dst_path: "{{ win_get_dst_path }}"
 
-- name: "Check if source path {{ win_src_path }} is a directory"
-  include_tasks: win_is_folder.yml
+- name: "Check if source path '{{ win_src_path }}' exists"
+  include_tasks: win_check_file_exist.yml
   vars:
-    win_is_folder_path: "{{ win_src_path }}"
+    win_check_file_exist_file: "{{ win_src_path }}"
 
-- name: "Get specified file from guest OS"
-  include_tasks: win_get_file.yml
-  vars:
-    win_get_file_src_path: "{{ win_src_path }}"
-    win_get_file_dst_path: "{{ win_dst_path }}"
-  when: not win_is_folder_result
+- name: "Source path '{{ win_src_path }}' not exist in guest OS"
+  ansible.builtin.debug:
+    msg: "Skip getting '{{ win_src_path }}' from guest OS to local since it does not exist."
+  when: not win_check_file_exist_result
 
-- name: "Get files or sub folders in specified folder"
-  include_tasks: win_get_folder.yml
-  vars:
-    win_get_folder_src_path: "{{ win_src_path }}"
-    win_get_folder_dst_path: "{{ win_dst_path }}"
-  when: win_is_folder_result
+- name: "Source path '{{ win_src_path }}' exists in guest OS"
+  when: win_check_file_exist_result
+  block:
+    - name: "Check if source path '{{ win_src_path }}' is a directory"
+      include_tasks: win_is_folder.yml
+      vars:
+        win_is_folder_path: "{{ win_src_path }}"
+
+    - name: "Get specified file from guest OS"
+      include_tasks: win_get_file.yml
+      vars:
+        win_get_file_src_path: "{{ win_src_path }}"
+        win_get_file_dst_path: "{{ win_dst_path }}"
+        win_get_file_ignore_error: true
+      when: not win_is_folder_result
+
+    - name: "Get files in specified folder from guest OS recursively"
+      include_tasks: win_get_folder.yml
```

Comment:
I saw `win_get_folder.yml` included `win_get_file_folder.yml` and `win_get_file_folder.yml` included `win_get_folder.yml`.  Won't the insider vars changing affect outsider tasks?

I think it's better to get a list of files' full paths under this folder and all its sub folders, then use `win_get_file.yml` to fetch them one by one.

Reply:
I removed these 2 utils files since they're not used in test cases. If there is new request then will add new tasks with `ansible.windows.win_find` module it supports 'get files recursively' now. Thanks.